src/data/abilities/apply-ab-attrs.ts

```diff
@@ -5,24 +5,48 @@ import { AbilityApplyMode } from "#enums/ability-apply-mode";
 import type { AbilityFilterOptions } from "#types/ability-filter-options";
 import { queueShowAbility } from "#utils/ability-utils";
 
+//#region ApplyAbAttrsResult
+
+interface AppliedAbAttr<TAttr extends AbAttr> {
+  /** The ability attribute */
+  attr: TAttr;
+  /** The (simulated) result of applying the attribute */
+  result: boolean;
```

Comment:
Again not fully sure about this name, something like `applies` or `applied` seems better to me.

Not entirely sure the comment is super clear either, if i understand correctly `result` here is the return value from the ability's `apply` function, which says this: 
```
   * @returns `true` if this attribute applies successfully. If {@linkcode showAbility} is enabled,
   * and this apply call is not simulated, returning `true` activates the ability's flyout
   * and {@linkcode getTriggerMessage | trigger message} (if applicable)
```
So all in all I'd say something like that
```suggestion
  /** Whether the ability was applied (or would have be applied, if the check wasn't simulated) */
  applied: boolean;
```

People more familiar with the ability code probably would have better insight to provide

Reply:
That's actually a good idea. This would also make it easier in my mind to name the interface something with `Result`. I simply adopted the name from the original function body